docs/docs-beta/docs/tutorial/pinecone.md

```diff
@@ -0,0 +1,311 @@
+---
+title: Pinecone
+description: Use Dagster to upload data into Pinecone
+last_update:
+  date: 2024-12-30
+  author: Dennis Hume
+---
+
+## Overview
+Many AI applications are data applications. Organizations want to leverage existing LLMs rather than build their own. But in order to take advantage of all the models that exist, you need to supplement them with your own data to produce more accurate and contextually aware results. We will demonstrate how to use Dagster to extract data, generate embeddings, and store the results within a vector database ([Pinecone](https://www.pinecone.io/)) which we can then use to power AI models to craft far more detailed answers.
+
+### Dagster Concepts
+- [resources]()
+- [run configurations]()
+
+### Services
+- [DuckDB]()
+- [Pinecone]()
+
+## Code
+![Pinecone asset graph](/images/tutorials/pinecone/pinecone_dag.png)
+
+### Setup
+All the code for this tutorial can be found at [project_dagster_pinecone]().
+
+Install the project dependencies:
+```
+pip install -e ".[dev]"
+```
+
+Run Dagster:
+```
+dagster dev
+```
+
+Open [http://localhost:3000](http://localhost:3000) in your browser.
+
+### Loading Data
+We will be working with review data from Goodreads. These reviews exist as a collection of JSON files categorized by different genres. We will focus on just the files for graphic novels to limit the size of the files we will process. Within this domain, the files we will be working with are `goodreads_books_comics_graphic.json.gz` and `goodreads_reviews_comics_graphic.json.gz`. Since the data is normalized across these two files, we will want to combine information before feeding it into our vector database.
+
+One way to handle preprocessing of the data is with [DuckDB](https://duckdb.org/). DuckDB is an in-process database, similar to SQLite, optimized for analytical workloads. We will start by creating two Dagster assets to load in the data. Each will load one of the files and create a DuckDB table (`graphic_novels` and `reviews`):
+```python
```

Comment:
Should we use this tutorial as an example of embedding the code with `CodeExample`?


Right now `docs_beta_snippets` is hard coded, but we could improve the component to take in a `prop` of the snippet location, and default to `docs_beta_snippets` if not provided.

```
    import(`!!raw-loader!/../../examples/docs_beta_snippets/docs_beta_snippets/${filePath}`)
```

https://github.com/dagster-io/dagster/blob/master/docs/docs-beta/src/components/CodeExample.tsx#L44